### DISCUSSION TITLE
Simple fade in animation of image

### DISCUSSION THREAD

**Discussion by marco-introini** [alpinejs/alpine#2623] [1. Help] [answered]
Hi everybody
I have a simple question but cannot figure out what the problem is.
I want to show an image when the page is loaded with a simple fade-in effect. I used x-transition in this way:

<div class="flex-col mt-5 mx-auto"
x-data="{ show : false }">

    <div class="flex w-fit mx-auto max-h-[800px]"
         x-show="show"
         id="immagine_full"
         x-transition
         x-transition:enter="transition border duration-300"
         x-transition:enter-start="opacity-10 border-0"
         x-transition:enter-end="opacity-100 border">
        <img class="border border-black object-contain"
             x-on:load="{ show=true }"
             src="{{ $img->path}}"
             alt=" {{$img->name }}">
    </div>
</div>

It works almost always. Almost. Sometimes after a reload or after the first load the show variable il still false and nothing is shown. It's driving me crazy. It seems randomly working
How do you manage a fade in effect onload?

Really thanks in advance

**Comment by SimoTod** [comment, marked as answer]
It's a race condition between the image and Alpine. If the load event is triggered before alpine starts, the variable will never change to true (it can happen, especially if the image is already in the browser cache).

**Reply by SimoTod** [reply]
You can try adding `x-init="if($el.complete) show=true"` to the image to cover the case where the image loads before alpine kicks in. You still need to keep the x-on to cover the other case.

**Reply by marco-introini** [reply]
Thanks a lot!
I had to use this:
x-init="if(document.getElementById('immagine_full').complete) show=true"
instead of $el

but now it seems to work

Thanks!!!

**Reply by SimoTod** [reply]
it's odd, that id is a property of the div tag in your example and complete should be a property of the image tag (https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/complete). Maybe keep an eye on it, it may still be not working.

**Reply by marco-introini** [reply]
Sorry, you're right. I also moved the id to the image. I forgot to paste it in this thread.
Thanks :)
